src/control/shaperecognizer/ShapeRecognizer.cpp

```diff
@@ -297,22 +297,30 @@ auto ShapeRecognizer::recognizePatterns(Stroke* stroke) -> ShapeRecognizerResult
 
         if (n == 1)  // current stroke is a line
         {
+            bool aligned = true;
             if (fabs(rs->angle) < SLANT_TOLERANCE)  // nearly horizontal
             {
                 rs->angle = 0.0;
                 rs->y1 = rs->y2 = rs->ycenter;
-            }
-            if (fabs(rs->angle) > M_PI / 2 - SLANT_TOLERANCE)  // nearly vertical
-            {
+            } else if (fabs(rs->angle) > M_PI / 2 - SLANT_TOLERANCE) {  // nearly vertical
                 rs->angle = (rs->angle > 0) ? (M_PI / 2) : (-M_PI / 2);
                 rs->x1 = rs->x2 = rs->xcenter;
+            } else {
+                aligned = false;
             }
 
             auto* s = new Stroke();
             s->applyStyleFrom(this->stroke);
 
-            s->addPoint(Point(rs->x1, rs->y1));
-            s->addPoint(Point(rs->x2, rs->y2));
+            if (aligned) {
+                s->addPoint(Point(rs->x1, rs->y1));
+                s->addPoint(Point(rs->x2, rs->y2));
+            } else {
+                auto points = stroke->getPoints();
+                s->addPoint(Point(points[0].x, points[0].y));
+                s->addPoint(Point(points[stroke->getPointCount() - 1].x, points[stroke->getPointCount() - 1].y));
```

Comment:
You can still simplify this:
```suggestion
                s->addPoint(points[0]);
                s->addPoint(points[stroke.size() - 1]);
```

Reply:
Thanks! I didn't notice Stroke::getPointVector, so I'm using that instead of Stroke::getPoints. Now I can just use vector::front and vector::back.